pkg/serviceimport/map.go

```diff
@@ -182,11 +184,22 @@ func (m *Map) Remove(serviceImport *mcsv1a1.ServiceImport) {
 		if len(remoteService.records) == 0 {
 			delete(m.svcMap, key)
 		} else if !remoteService.isHeadless {
-			remoteService.buildClusterInfoQueue()
+			remoteService.resetLoadBalancing()
 		}
 	}
 }
 
+func getServiceWeightFrom(annotation map[string]string) int64 {
+	if val, ok := annotation["weight"]; ok {
```

Comment:
I think we should qualify the name to avoid potential conflicts.

```suggestion
	if val, ok := annotation["load-balancer.submariner.io/weight"]; ok {
```